cache.go

```diff
@@ -139,11 +140,32 @@ func (c *cache) remove(name string) error {
 	return nil
 }
 
+func (c *cache) monitor(tick time.Duration) {
+	ticker := time.NewTicker(tick)
+	for {
+		select {
+		case <-ticker.C:
+			c.mu.RLock()
+			defer c.mu.RUnlock()
+			for _, entry := range c.entries {
+				if entry.timeToUpdate() {
+					go func() {
+						err := entry.refreshResponse()
+						if err != nil {
+							entry.log.Err("Failed to refresh response: %s", err)
```

Comment:
Why not do this inside refreshResponse (or a wrapper function)? Also, then you can use `e.err` (which you can do here too, but seems more natural there).